### PR TITLE
Fix regression of ImageBitmap in Safari

### LINKED ISSUE #324: Regression in safari on Observable HQ in maplibre-gl@2.0.0-pre.1 ReferenceError: Can't find variable: ImageBitmap
**maplibre-gl-js version**:

maplibre-gl@2.0.0-pre.1

**browser**:

Safari

### Steps to Trigger Behavior

 1. Visit https://observablehq.com/@jimmyg/gmaplibre-regression in the Safari browser
 2. You'll see `ReferenceError: Can't find variable: ImageBitmap`
 3. If you change the import at the top to `maplibregl = await require("maplibre-gl@1.15.2")` it works fine

When inspecting the debug console I see:

```
worker ↑
Object

errors: Map {15 => {message: "Can't find variable: ImageBitmap", stack: "update@https://cdn.jsdelivr.net/npm/maplibre-gl@2.…gl.js:35:318889↵↵promiseReactionJob@[native code]", input: undefined, name: "ReferenceError"}} (1)

type: "errors"

Object Prototype
```

Safari doesn't support `ImageBitmap` so this suggests to me that perhaps the code that does feature detection isn't working correctly as it tries to use `ImageBitmap` in the worker. It works fine in Chrome.

Of course this could just be an issue in the ObservableHQ environment, I don't know enough about the maplibre-gl-js project to know.

I don't consider this a serious issue for me at the mo, but wanted to flag it in case it points to a genuine problem in the 2.0.0 branch and later effects that release.

### Link to Demonstration

I've ported the first example in the docs from https://maplibre.org/maplibre-gl-js-docs/example/simple-map/ to ObservableHQ syntax:

https://observablehq.com/@jimmyg/gmaplibre-regression

### Expected Behavior

The map should display in Safari (it works fine Chrome).

### Actual Behavior

`ReferenceError: Can't find variable: ImageBitmap`

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist

Fix issue in worker for ImageBitmap not defined in safari - #324
Also this allowed me to learn about narrowing typing using the `val is Type` as can be seen here:
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates
Which is very cool and allowed removing of some unwanted `as` statements :-)

 - [x] confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes

### CODE REVIEW

File: src/util/web_worker_transfer.ts [hunk 1]
```diff
@@ -135,14 +131,14 @@ export function serialize(input: unknown, transferables?: Array<Transferable> |
...
-            transferables.push(((input as any as ArrayBuffer)));
+            transferables.push(((input)));
         }
         return input as Serialized;
     }
 
     if (isImageBitmap(input)) {
         if (transferables) {
-            transferables.push(((input as any as ImageBitmap)));
+            transferables.push(((input)));
```
Comment (wipfli) [inline src/util/web_worker_transfer.ts:141]: Remove parentheses

File: src/util/web_worker_transfer.ts [hunk 2]
```diff
@@ -135,14 +131,14 @@ export function serialize(input: unknown, transferables?: Array<Transferable> |
 
     if (isArrayBuffer(input)) {
         if (transferables) {
-            transferables.push(((input as any as ArrayBuffer)));
+            transferables.push(((input)));
```
Comment (wipfli) [inline src/util/web_worker_transfer.ts:134]: Remove parentheses

File: src/util/util.ts [hunk 3]
```diff
@@ -500,3 +500,7 @@ export function b64DecodeUnicode(str: string) {
         return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2); //eslint-disable-line
     }).join(''));
 }
+
+export function isImageBitmap(image: any): image is ImageBitmap {
+    return typeof ImageBitmap !== 'undefined' && image instanceof ImageBitmap;
+}
```
Comment (wipfli) [inline src/util/util.ts:506]: Should we have done the same with `OffscreenCanvas`? https://github.com/maplibre/maplibre-gl-js/pull/308

File: src/render/texture.ts [hunk 4]
```diff
@@ -12,7 +13,7 @@ type EmptyImage = {
 };
 
 type DataTextureImage = RGBAImage | AlphaImage | EmptyImage;
-export type TextureImage = HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | ImageData | ImageBitmap | DataTextureImage;
+export type TextureImage = TexImageSource | DataTextureImage;
```
Comment (wipfli) [inline src/render/texture.ts:16]: Please comment why you changed this.

Comment (HarelM) [reply src/render/texture.ts:16]: It is almost the same and this is what webgl expects to receive. So basically less types defined by us.

File: src/util/util.ts (hunk 3 above)
Comment (HarelM) [reply src/util/util.ts:506]: No, since off canvas is used once while this is used multiple times...

File: src/util/web_worker_transfer.ts (hunk 2 above)
Comment (HarelM) [reply src/util/web_worker_transfer.ts:134]: I will, thanks.

File: src/util/web_worker_transfer.ts (hunk 1 above)
Comment (HarelM) [reply src/util/web_worker_transfer.ts:141]: Who placed all of these??? :-)

Comment (wipfli) [reply src/util/web_worker_transfer.ts:141]: My theory is that this library was transformed from some flow predecessor to flow with an automatic tool and now we transformed it again from flow to typescript with an automatic tool. Maybe in 20 years we will have 5 or 10 redundant parentheses...

**Review by wipfli — APPROVED** [review]

**Comment by HarelM** [comment]
I've fixed according to code review changes and bumped to pre.3 in order to create another release for testing.

File: src/util/util.ts (hunk 3 above)
Comment (HarelM) [reply src/util/util.ts:506]: I think I now fully understand what you asked, it is a good idea to move the offscreencanvas to the same file in theory, but offscreen canvas is cached due to the fact that it's "expensive" to calculate it, so there's a variable to hold the calculated cached value. In this case I think it's better to have it in a dedicated file...

**Pull request merged**